Sources/Core/Spec+Context.swift

```diff
@@ -0,0 +1,51 @@
+//  Spec+Context.swift
+
+import Foundation
+
+extension Spec.Product {
+    func makeContext() -> [String: Any] {
+        [
+            "name": name,
+            "productType": productType.rawValue,
+            "targets": targets
+        ]
+    }
+}
+
+extension Spec.RemoteDependency {
+    func makeContext() -> [String: Any] {
+        var retVal = [
+            "name": name,
+            "url": url
+        ]
+        if let ref {
+            switch ref {
+            case .branch(let value):
+                retVal["branch"] = value
+            case .revision(let value):
+                retVal["revision"] = value
+            case .version(let value):
+                retVal["version"] = value
+            }
+        }
+        return retVal.compactMapValues { $0 }
+    }
+}
+
+extension Spec {
+    func makeContext() -> [String: Any] {
+        let values: [String: Any?] = [
+            "package_name": name,
+            "platforms": platforms,
+            "local_dependencies": localDependencies,
+            "remote_dependencies": remoteDependencies.map{ $0.makeContext() },
```

Comment:
minor: 
```suggestion
            "remote_dependencies": remoteDependencies.map { $0.makeContext() },
```
tot follow a style that you have